analyses/infercnv-consensus-cell-type/notebooks/01_ewings-consensus-cell-types.Rmd

```diff
@@ -0,0 +1,186 @@
+
+## Setup
+
+```{r}
+options(readr.show_col_types = FALSE)
+```
+
+### Paths
+
+#### Base directories
+
+
+```{r base paths}
+# The base path for the OpenScPCA repository, found by its (hidden) .git directory
+repository_base <- rprojroot::find_root(rprojroot::is_git_root)
+
+# The current results directory, found within the repository base directory
+result_dir <- file.path(repository_base, "data", "current", "results")
+
+# Path to cell-type-ewings results, which also contains consensus annotations
+ewings_dir <- file.path(result_dir, "cell-type-ewings", "SCPCP000015")
+
+# Path to immune cell list
+immune_url <- "https://raw.githubusercontent.com/AlexsLemonade/OpenScPCA-analysis/refs/heads/main/analyses/cell-type-consensus/references/consensus-immune-cell-types.tsv"
+```
+
+#### Input and output files
+
+Set paths to input and output directories and files in the chunk below.
+
+```{r paths}
+# Input files
+celltype_files <- list.files(
+  path = ewings_dir,
+  pattern = "_ewing-celltype-assignments\\.tsv$",
+  recursive = TRUE,
+  full.names = TRUE
+)
+```
+
+## Analysis 
+
+```{r}
+immune_celltypes <- readr::read_tsv(immune_url) |>
+  dplyr::pull(consensus_annotation)
+
+celltype_df <- celltype_files |>
+  purrr::map(readr::read_tsv) |>
+  purrr::list_rbind() |>
+  dplyr::select(library_id, barcodes, ewing_annotation, consensus_annotation) |>
+  dplyr::mutate(
+    category = dplyr::case_when(
+      consensus_annotation == "Unknown" ~ "Unknown",
+      consensus_annotation %in% immune_celltypes ~ "immune",
+      .default = "other"
+    )
+  )
+```
+
+
+```{r}
+category_count_df <- celltype_df |>
+  dplyr::count(library_id, category) |>
+  tidyr::pivot_wider(
+    names_from = category,
+    values_from = n,
+    values_fill = 0
+  )
+
+# this way we can print the full table at once; RMarkdown will only show 10 rows at once
+knitr::kable(category_count_df)
+```
+It seems most immune cells are concentrated in 2-3 samples, and most samples don't have any immune cells.
+
+How many immune cells are there total?
+
+```{r}
+total_immune <- sum(category_count_df$immune)
+total_immune
+```
+
+`inferCNV` authors [recommend that at least 100 cells](https://github.com/broadinstitute/infercnv/issues/188) be used for a "normal" reference, so we are above that threshold.
+
+If we were to use all of these, what fraction is that per sample?
+We'll just divide the total number of cells by this count, not taking into consideration how many immune cells are actually in that sample, to get a rough sense of how many immune cells there are relative to all.
+
+```{r}
+celltype_df |>
+  dplyr::count(library_id, name = "total_cells") |>
+  dplyr::mutate(fraction_immune = round(total_immune / total_cells, 2)) |>
+  knitr::kable()
+```
+
+For the low-quality sample `SCPCL001111` this fraction isn't very meaningful since there are fewer cells in the sample than there are pooled immune cells.
```

Comment:
We should probably just exclude this sample from the analysis?

Reply:
💯